### ISSUE TITLE
Add support for docket history reports to Juriscraper, API, and RECAP extension

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#766]
Right now the extensions are sending docket history report HTML to the server as if it's a docket. I didn't catch that this was happening, so the server is barfing whenever it gets one of these. Instead, we should recognize them for what they are, and do the right thing. There's a lot of good data in here we should be capturing.

So:
 - Add support for these to Juriscraper.
 - Add a new upload_type to the API on courtlistener
 - Tweak the extension so that it uploads these with the correct upload_type instead of as dockets

Not sure how I missed this until now, but there don't seem to be tests for it on the original RECAP server, so I didn't catch that it was something it supported. There's a lot of code on that server.

This needs to be fixed pretty quick, but the good news is that we're not losing data. It's just that it's not getting merged into CourtListener. Once the parser is done, we can use the existing collection of failed docket parses as a solid test set.

In terms of the scale of the issue, so far we've got 552 items that look like they're doc history reports that haven't been parsed correctly. That's out of about 4k dockets so far.

**Comment by mlissner** [comment]
OK, when we push out RECAP 1.1.8, it'll push these items to the server with a upload_type of 4, which now corresponds to `DOCKET_HISTORY_REPORT`. For the moment, that's just stubbed out and we're just saving the items. Got to do the last piece and get the parser/merger coded up.

**Comment by mlissner** [comment]
All set here. Currently processing the backlog of about 10k items. Looks like we'll add a few hundred thousand items to the DB when we process the backlog.

**Comment by mlissner** [comment]
Grand total after reprocessing everything since November:

 - 2960 new dockets, 5800 modified
 - 248k new docket entries, with...
 - 258k new RECAP Document objects, with...
 - 5k documents with associated PDFs that were previously orphaned without metadata

All in all, a pretty great addition. Surprising how much this report is used.
